### PR TITLE
Unit tests for Cov2d Denoiser 

### PR DESCRIPTION
**Pull request by junchaoxia**
Closing issue #293.

### CODE REVIEW

File: tests/test_covar2d_denoiser.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            )
+        )
+
+    def testMSE(self):
+        # need larger numbers of images and higher resolution for good MSE
+        img_size = 64
+        num_imgs = 1024
+
+        # Create filters
+        ctf_filters = [
```
Comment (vineetbansal) [inline tests/test_covar2d_denoiser.py]: Should we be using the same filters as in the constructor (in which case, maybe set to an attribute?)

Comment (junchaoxia) [reply tests/test_covar2d_denoiser.py]: Done in commit ed303d3.

File: tests/test_covar2d_denoiser.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            "precision": self.dtype,
+        }
+
+    def blk_diag_allclose(self, blk_diag_a, blk_diag_b, atol=1e-8):
+        close = True
+        for blk_a, blk_b in zip(blk_diag_a, blk_diag_b):
+            close = close and np.allclose(blk_a, blk_b, atol=atol)
+        return close
+
+    def testMean(self):
```
Comment (janden) [inline tests/test_covar2d_denoiser.py]: Remove all tests except `testMSE`.

File: tests/test_covar2d_denoiser.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            offsets=0.0,
+            amplitudes=1.0,
+            dtype=self.dtype,
+            noise_filter=self.noise_filter,
+        )
+        imgs_clean = sim.projections()
+
+        # Specify the fast FB basis method for expending the 2D images
+        ffbbasis = FFBBasis2D((img_size, img_size), dtype=self.dtype)
+        denoisor = DenoiserCov2D(sim, ffbbasis, self.noise_var)
```
Comment (janden) [inline tests/test_covar2d_denoiser.py]: `denoiser`

**Review by janden — CHANGES_REQUESTED** [review]

File: tests/test_covar2d_denoiser.py (hunk 2 above)
Comment (junchaoxia) [reply tests/test_covar2d_denoiser.py]: Done in commit 2c68610.

File: tests/test_covar2d_denoiser.py (hunk 3 above)
Comment (junchaoxia) [reply tests/test_covar2d_denoiser.py]: Done in commit 2c68610.

**Review by janden — APPROVED** [review]

**Pull request merged**